frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -112,12 +315,13 @@ impl<T: Config> Pallet<T> {
 		// Any unprocessed solution is pointless to even consider. Feasible or malicious,
 		// they didn't end up being used. Unreserve the bonds.
 		let discarded = all_submissions.len();
-		for not_processed in all_submissions {
-			let SignedSubmission { who, deposit, .. } = not_processed;
+		for SignedSubmission { who, deposit, .. } in all_submissions.drain() {
```

Comment:
When we drain with our btreemap (e.g. `SignedSubmissionsMap::<T>::drain()`), do we load/decode all the entries from trie into memory at once? Or does each iteration load/decode a single entry? (Maybe a better question is where could I find the `drain` impl? 😬)

Also, does the `SignedSubmission` get dropped from memory once we go to the next loop iteration?

Reply:
This calls the `fn drain` method defined on line 167. From there, it delegates to `SignedSubmissionMap::<T>::drain()`, which is an implementation of `IterableStorageMap::drain` which loads/decodes a single entry at a time.

And yes, each `SignedSubmission` gets dropped at the end of its loop iteration.

Reply:
Good point, but this is safe since as mentioned we allocate one by one. 

> Maybe a better question is where could I find the drain impl


https://crates.parity.io/src/frame_support/storage/generator/map.rs.html#135-195